Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -653,6 +653,26 @@ public static AsyncFunctionAssertions Should<T>(this Func<Task<T>> action)
             return new AsyncFunctionAssertions(action.ExecuteInDefaultSynchronizationContext, extractor);
         }
 
+        /// <summary>
+        /// Returns a <see cref="TaskAssertions"/> object that can be used to assert the
+        /// current <see cref="Task"/> .
+        /// </summary>
+        [Pure]
+        public static TaskAssertions Should(this Task task)
+        {
+            return new TaskAssertions(task, new TaskTimer());
```

Comment:
🤔 If we assert on a `Task` hasn't the execution already started? 
That would mean we aren't measuring the execution time from the beginning of the execution.

Reply:
Yes, that may happen.
Do you have any suggestion?

Reply:
How about reusing the existing `Func<Task>` and `Func<Task<T>>` extension methods?

Reply:
Could you be a little more specific, please?
Further I think that such a problem cannot be solved completely precisely: The timing starts either before the start of the task and measures too much time. Or the measurement starts after the task and measures too much time.

Reply:
Of course, I would reuse these two existing overloads of `Should` that takes `Func<Task>` and `Func<Task<T>>`
```c#
public static AsyncFunctionAssertions Should(this Func<Task> action)
public static AsyncFunctionAssertions Should<T>(this Func<Task<T>> action)
```

Measuring execution times is inherently difficult, for unit tests without warm-up, is downright unreliable.
That requires dedicated benchmark tools such as [BenchmarkDotNet](https://benchmarkdotnet.org/).
E.g. the order in which the unit tests are executed now matters, due to GC, JITting, threads, etc.
As we can only execute a single statement at a time, we cannot start measuring at the exact same time the execution starts.
All other places (I hope so) we start the clock before starting the execution.
See e.g. [ExecutionTimeAssertions.cs](https://github.com/fluentassertions/fluentassertions/blob/master/Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs)

Reply:
Within this kind of extension we have no chance to start timing before starting the `Task`. It is an extension for the task (instance) itself.
We can either start timing after task is running or we can stop here. Isn't it?

Reply:
My thought is to avoid these two overloads of `Should` and use the existing ones.
The reason is to avoid fragmenting the API.
Then `CompleteWithin` could be added to `AsyncFunctionAssertions`?

Reply:
I agree with @jnyrup. And it's not just about _when_ your timer starts running. It's purely to ensure we have a consistent API. After reading our response on Slack, I really hope you're willing to finalize this and the other comments. I'd love to have this in the next release.